src/debugAdapter/goDebug.ts

```diff
@@ -12,6 +12,7 @@ import { basename, dirname, extname } from 'path';
 import { spawn, ChildProcess, execSync, spawnSync, execFile } from 'child_process';
 import { Client, RPCConnection } from 'json-rpc2';
 import { parseEnvFile, getBinPathWithPreferredGopath, resolveHomeDir, getInferredGopath, getCurrentGoWorkspaceFromGOPATH, envPath, fixDriveCasingInWindows } from '../goPath';
+import { TempFileProvider } from '../util';
```

Comment:
The debug adapter runs in a separate process and does not have access to anything from the `vscode` module. Therefore a dependency here on `util` will fail as util imports from `vscode`.

The tmp dir is used in the debug adapter to create the trace files. Since this is not done by default and the user sets explicitly to debug an issue with the debugger, I believe we are ok with creating a new file under `os.tmpdir()`

Reply:
👍